### ISSUE TITLE
WebView errorCode: 101 on some sites, after initially loading successfully

### ISSUE THREAD

**Issue by mschipperheyn** [facebook/react-native#7839]
On some urls I get a weird 101 error after successfully loading the page (which replaces the loaded page with the error page).

URLs that work, e.g.: http://www.redbull.com
URLs that don't work, e.g.: http://www.cnn.com

```
domain: "WebKitErrorDomain"
errorCode: 101
Description: The URL can't be shown
```

The code I use is this:

```
 <WebView
        source={{ uri: url}}
        scalesPageToFit={true}
        startInLoadingState={true}
        renderError={this._renderError.bind(this)}
        renderLoading={this._renderProgress.bind(this)}
        onLoad={this.onLoad.bind(this)}
        style={localStyle.browserModal}
        />
```

The WebView is shown in a Modal.

I do use a setState in onLoad function, triggering a re-render.

**Comment by charpeni** [comment]
What version of React Native are you using ? This occurs on Android, iOS, or both?

**Comment by jaygarcia** [comment]
That's because CNN is shit. 

Kidding. Can't help without specifics, @mschipperheyn

**Comment by fungilation** [comment]
I'm seeing the same issue.

RN 0.25.1, loading ```http://www.bloomberg.com/quote/SUNEQ:US``` in WebView.

**Comment by longsangstan** [comment]
I have the same issue.
Some url work, and some don't - like bloomberg as mentioned by @fungilation

**Comment by fungilation** [comment]
RN Webview really need modernizing to use WKWebView instead. For every reason of performance, security and support.

**Comment by ronniegong** [comment]
same error here
how to solve this?

**Comment by willyyang** [comment]
same issue as well, has anyone resolved this?

**Comment by donedgardo** [comment]
👍 same here.

**Comment by adrianotadao** [comment]
I have a similar error, but I'm receiving the error 102. =l

I'm trying to open a URL that will redirect me to a `customDomain://`... 😢

**Comment by fungilation** [comment]
There should be better handling of redirects to unknown URL handlers. Like
stop loading such and log with the error

**Comment by messuti-edd** [comment]
Having this issue too with this URL https://www.bloomberg.com/news/articles/2016-10-26/snapchat-seeks-to-raise-as-much-as-4-billion-in-ipo :(

**Comment by adrianotadao** [comment]
I solved the error 102, doing it:

```javascript
<WebView
  domStorageEnabled={ true }
  javaScriptEnabled={ true }
  renderError={ (e) => {
    if (e === 'WebKitErrorDomain') {
      return
    }
  }}
  source={{ uri: "https://99f9e623.ngrok.io/auth/sign_in_with_slack?source=app" }}
  style={ styles.container }
/>
```

I now that isn't nice, but it works for me.

**Comment by messutied-pf** [comment]
Thanks!, I will try this :)

**Comment by lacker** [comment]
It seems like this is a duplicate of https://github.com/facebook/react-native/issues/7839 so I am going to close this issue, and post the workaround over there.

**Comment by fungilation** [comment]
@lacker this is #7839. What duplicate are you talking about?

**Comment by lacker** [comment]
Arrr, sorry I was going through a bunch of things in a batch and I must have gotten confused. But I looked around more and the duplicate issue appears to be https://github.com/facebook/react-native/issues/9037 . I am still going to close this issue. Sorry for the churn, my bad!

**Comment by alexkasemir** [comment]
So I believe that the workaround that @adrianotadao posted was for a 
```
domain: "WebKitErrorDomain"
errorCode: 102
```
and not one with the `errorCode:101`. I do not believe that the workaround fixes this issue for #7839 or #9037

**Comment by lacker** [comment]
Ok I am going to reopen this and remove the comments about error 102 to make it clear that error 102 is being discussed in https://github.com/facebook/react-native/issues/9037 not in this issue.

**Comment by alexkasemir** [comment]
@lacker Error 101 is being discussed in #9037 as well not Error 102

**Comment by lacker** [comment]
Ah, you are right. So the remaining comments here are actually a dupe of https://github.com/facebook/react-native/issues/9037 after all. OK, I think unfortunately I have mauled this issue into confusion because I mishandled the whole multiple-issues-reporting-some-overlapping-subset-of-distinct-problems thing. I'm going to re-close it, partly out of despair, partly because the remaining information is duplicated anyway. If there is something we are missing with the current set of open issues, perhaps it would be best to open a new issue and be very precise in the title about which bug it is reporting.
